pinecone/index.py

```diff
@@ -70,31 +115,169 @@ def _vector_transform(item):
+                     See https://www.pinecone.io/docs/metadata-filtering/.. [optional]
+
+      Keyword Args:
+        Supports OpenAPI client keyword arguments. See pinecone.core.client.models.DeleteRequest for more details.
+
+
+        Returns: An empty dictionary if the delete operation was successful.
+
+        Call Examples:
+            index.delete(ids=['id1', 'id2'], namespace='my_namespace')
+            index.delete(delete_all=True, namespace='my_namespace')
+            index.delete(filter={'key': 'value'}, namespace='my_namespace')
+        """
         _check_type = kwargs.pop('_check_type', False)
+        args_dict = self._parse_args_to_dict([('ids', ids),
+                                              ('delete_all', delete_all),
+                                              ('namespace', namespace),
+                                              ('filter', filter)])
+
         return self._vector_api.delete(
             DeleteRequest(
-                *args,
-                **{k: v for k, v in kwargs.items() if k not in _OPENAPI_ENDPOINT_PARAMS},
+                **args_dict,
+                **{k: v for k, v in kwargs.items() if k not in _OPENAPI_ENDPOINT_PARAMS and v is not None},
                 _check_type=_check_type
             ),
             **{k: v for k, v in kwargs.items() if k in _OPENAPI_ENDPOINT_PARAMS}
         )
 
     @validate_and_convert_errors
-    def fetch(self, *args, **kwargs):
-        return self._vector_api.fetch(*args, **kwargs)
+    def fetch(self,
+              ids: List[str],
+              namespace: Optional[str] = None,
+              **kwargs) -> FetchResponse:
+        """
+        The fetch operation looks up and returns vectors, by ID, from a single namespace.
+        The returned vectors include the vector data and/or metadata.
+
+        API reference: https://docs.pinecone.io/reference/fetch
+
+        Args:
+            ids (List[str]): The vector IDs to fetch.
+            namespace (str): The namespace to fetch vectors from.
+                             If not specified, the default namespace is used. [optional]
+        Keyword Args:
+        Supports OpenAPI client keyword arguments. See pinecone.core.client.models.FetchResponse for more details.
+
+
+        Returns: FetchResponse object which contains the list of Vector objects, and namespace name.
+
+        Call Examples:
+            index.fetch(ids=['id1', 'id2'], namespace='my_namespace')
+            index.fetch(ids=['id1', 'id2'])
+        """
+        args_dict = self._parse_args_to_dict([('namespace', namespace)])
+        return self._vector_api.fetch(ids=ids, **args_dict, **kwargs)
 
     @validate_and_convert_errors
-    def query(self, vector=[], id='', queries=[], **kwargs):
+    def query(self,
+              vector: Optional[List[float]] = None,
+              id: Optional[str] = None,
+              queries: Optional[Union[List[QueryVector], List[Tuple]]] = None,
```

Comment:
this is deprecated, are we keeping it only for backward compatibility?

Reply:
yes, we don't want to do anything in this version that might break code for someone